salt/key.py

```diff
@@ -474,7 +489,10 @@ def name_match(self, match, full=False):
         ret = {}
         if ',' in match and isinstance(match, six.string_types):
             match = match.split(',')
+        selected_state = self._get_selected_state(self.opts['state'])
         for status, keys in six.iteritems(matches):
+            if selected_state != '' and status != selected_state:
```

Comment:
Could you please replace `if selected_state != ''` with `if selected_state`?

Reply:
Fixed.